apps/backoffice-v2/src/pages/Statistics/components/PortfolioRiskStatistics/hooks/usePortfolioRiskStatisticsLogic/usePortfolioRiskStatisticsLogic.tsx

```diff
@@ -5,82 +5,74 @@ import {
   riskLevelToBackgroundColor,
   riskLevelToFillColor,
 } from '@/pages/Statistics/components/PortfolioRiskStatistics/constants';
+import { StatisticsOutputSchema } from '@/pages/Statistics/statistics.query';
+import { z } from 'zod';
 
-export const usePortfolioRiskStatisticsLogic = () => {
+export const usePortfolioRiskStatisticsLogic = ({
+  data,
+}: {
+  data: z.infer<typeof StatisticsOutputSchema>;
+}) => {
   const [parent] = useAutoAnimate<HTMLTableSectionElement>();
-  const [sorting, setSorting] = useState<SortDirection>('desc');
-  const onSort = useCallback(
+  const [riskIndicatorsSorting, setRiskIndicatorsSorting] = useState<SortDirection>('desc');
+  const onSortRiskIndicators = useCallback(
     (sort: SortDirection) => () => {
-      setSorting(sort);
+      setRiskIndicatorsSorting(sort);
     },
     [],
   );
-  const sortedData = useMemo(
-    () =>
-      [
-        { riskType: 'Scam and fraud', amount: 15 },
-        { riskType: 'IP Rights Infringement', amount: 12 },
-        { riskType: 'Missing Terms and Conditions', amount: 10 },
-        { riskType: 'Counterfeit Goods', amount: 8 },
-        { riskType: 'Sanctions', amount: 4 },
-      ]
-        ?.slice()
-        .sort((a, b) => {
-          if (sorting === 'asc') {
-            return a.amount - b.amount;
-          }
-
-          return b.amount - a.amount;
-        }),
-    [sorting],
-  );
+  const totalRiskIndicators = data.riskIndicators.reduce((acc, curr) => acc + curr.count, 0);
+  const riskIndicators = useMemo(() => {
+    return data.riskIndicators
+      .sort((a, b) => {
+        if (riskIndicatorsSorting === 'asc') {
+          return a.count - b.count;
+        }
+        return b.count - a.count;
+      })
+      .slice(0, 5);
```

Comment:
Slice was intentionally placed before sort since sort mutates and slice returns a new array. Last time I checked I could not use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted in the backoffice.